drivers/sensor/tmp116/tmp116.c

```diff
@@ -112,39 +122,77 @@ static int tmp116_channel_get(const struct device *dev,
 	return 0;
 }
 
+static int tmp116_attr_set(const struct device *dev,
+			   enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	int16_t value, id;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_OFFSET:
+		tmp116_device_id_check(dev, &id);
```

Comment:
I think it would be better to cache device id in device data, instead of trying to fetch device id whenever SENSOR_ATTR_OFFSET is being changed.

Reply:
I added a variable to store the id in device data.